server/server.go

```diff
@@ -325,9 +327,223 @@ type stats struct {
 	outMsgs       int64
 	inBytes       int64
 	outBytes      int64
+	sentBytes     int64 // Total of what is sent out, include all protocols
 	slowConsumers int64
 }
 
+// This is used by tests so we can run all server tests with a default route
+// compression mode. For instance:
+// go test -race -v ./server -cluster_compression=fast
+var testDefaultClusterCompression string
+
+// Compression modes.
+const (
+	CompressionNotSupported   = "not supported"
+	CompressionOff            = "off"
+	CompressionAccept         = "accept"
+	CompressionS2Auto         = "s2_auto"
+	CompressionS2Uncompressed = "s2_uncompressed"
+	CompressionS2Fast         = "s2_fast"
+	CompressionS2Better       = "s2_better"
+	CompressionS2Best         = "s2_best"
+)
+
+// defaultCompressionS2AutoRTTThresholds is the default of RTT thresholds for
+// the CompressionS2Auto mode.
+var defaultCompressionS2AutoRTTThresholds = []time.Duration{
```

Comment:
If I set best on my side, what does other side do?

Reason I am asking is for asymmetrical routes, like home internet where downstream and upstream is different. So I would like to set mine to best but have other side sending to me with say fast.

Reply:
Compression levels are asymmetric, meaning that you can have S1 chose "best" and the other side "fast". If one side has "accept", then it will use the same than given by the other side, but otherwise, each side can select the one they want. So in short, yes, it works as you would want/described above.